framework/src/main/java/org/checkerframework/framework/source/CompositeChecker.java

```diff
@@ -0,0 +1,756 @@
+   *
+   * <p>Does not need to be initialized to null or an empty list because it is always initialized
+   * via calls to {@link #instantiateSubcheckers}.
+   */
+  // Set to non-null when subcheckers is.
+  protected @MonotonicNonNull List<BaseTypeChecker> immediateSubcheckers = null;
+
+  /** Supported options for this checker. */
+  protected @MonotonicNonNull Set<String> supportedOptions = null;
+
+  /** Options passed to this checker. */
+  protected @MonotonicNonNull Map<String, String> options = null;
+
+  /**
+   * TreePathCacher to share between instances. Initialized either in getTreePathCacher (which is
+   * also called from {@link #instantiateSubcheckers}).
+   */
+  protected TreePathCacher treePathCacher = null;
+
+  /**
+   * The list of suppress warnings prefixes supported by this checker or any of its subcheckers
+   * (including indirect subcheckers). Do not access this field directly; instead, use {@link
+   * #getSuppressWarningsPrefixesOfSubcheckers}.
+   */
+  protected @MonotonicNonNull Collection<String> suppressWarningsPrefixesOfSubcheckers = null;
+
+  /** True if -AwarnUnneededSuppressions was supplied on the command line. */
+  // Not final because it is set in `init()`.
+  protected boolean warnUnneededSuppressions;
+
+  @Override
+  public void setRoot(CompilationUnitTree newRoot) {
+    super.setRoot(newRoot);
+    if (parentChecker == null) {
+      // Only clear the path cache if this is the main checker.
+      treePathCacher.clear();
+    }
+  }
+
+  /**
+   * Returns the set of subchecker classes on which this checker depends. Returns an empty set if
+   * this checker does not depend on any others.
+   *
+   * <p>Subclasses should override this method to specify subcheckers. If they do so, they should
+   * call the super implementation of this method and add dependencies to the returned set so that
+   * checkers required for reflection resolution are included if reflection resolution is requested.
+   *
+   * <p>Each subchecker of this checker may also depend on other checkers. If this checker and one
+   * of its subcheckers both depend on a third checker, that checker will only be instantiated once.
+   *
+   * <p>Though each checker is run on a whole compilation unit before the next checker is run, error
+   * and warning messages are collected and sorted based on the location in the source file before
+   * being printed. (See {@link #printOrStoreMessage(Diagnostic.Kind, String, Tree,
+   * CompilationUnitTree)}.)
+   *
+   * <p>WARNING: Circular dependencies are not supported nor do checkers verify that their
+   * dependencies are not circular. Make sure no circular dependencies are created when overriding
+   * this method. (In other words, if checker A depends on checker B, checker B cannot depend on
+   * checker A.)
+   *
+   * <p>This method is protected so it can be overridden, but it should only be called internally by
+   * the BaseTypeChecker.
```

Comment:
The references in the documentation of this class' methods that mention BaseTypeChecker need to be updated (there are several places in this class with this issue).